includes/RestApi/Themes/PatternsController.php

```diff
@@ -143,4 +176,71 @@ public function set_pattern( \WP_REST_Request $request ) {
 
 		 return Patterns::set_theme_step_patterns( $step, $slug );
 	}
+
+	/**
+      * Retrieves all the header patterns.
+      *
+      * @return \WP_Rest_Response|\WP_Error
+      */
+	  public function get_all_header_patterns()
+	  {
+		   $slug_keyword = 'header';
+ 
+		   $header_patterns = Patterns::get_all_patterns_for_slug($slug_keyword);
+		   if (!$header_patterns) {
+				return new \WP_Error(
+					 'no_patterns_found',
+					 __('No Patterns Found.'),
+					 array('status' => 404)
+				);
+		   }
+ 
+		   return new \WP_REST_Response(
+				$header_patterns
+		   );
+	  }
+	  
+	/**
+      * Retrieves deafult the header pattern.
+      *
+      * @return \WP_Rest_Response|\WP_Error
+      */
+      public function get_default_header_pattern()
+      {
+           $header_pattern = Patterns::get_default_header();
+           if (!$header_pattern) {
+                return new \WP_Error(
+                     'no_default_pattern_found',
+                     __('No Default Pattern Found.'),
+                     array('status' => 404)
+                );
+           }
+
+           return new \WP_REST_Response(
+                $header_pattern
+           );
+      }
+
+	  /**
+      * Retrieves pattern for specific header pattern.
+      *
+      * @return \WP_Rest_Response|\WP_Error
+      */
+	  public function get_header_pattern_preview(\WP_REST_Request $request)
+	  {
+		   $slug   = $request->get_param('slug');
```

Comment:
Can we add an args to register route like being used in "public function get_pattern_args()" so as to facilitate the case where we don't pass the slug as the parameter.
Please make sure all the endpoints have this args function to make them fail safe as right now this function returns a Body and Footer when no slug is provided rather than an error message for reminding to send the slug.